### PR TITLE
Attribute based equivalence for commands

### PR DESCRIPTION
**Pull request by rfeistenauer**
I have implemented a command equivalence, that does not just compare the object, but its attributes. 
e.g currently: 
```
Ptp() == Ptp() 
False
```
with this PR its True:
```
Ptp() == Ptp()
True
Ptp() == Lin()
False
Ptp(goal=[1, 2]) == Ptp(goal=[1, 2])
True
```

Needed this to greatly reduce the effort for comparison in some tests.
Are there any arguments against including this as a feature into the api?

### CODE REVIEW

File: pilz_robot_programming/src/pilz_robot_programming/commands.py
Comment (ct2034) [inline pilz_robot_programming/src/pilz_robot_programming/commands.py]:
```suggestion
        else:
            return NotImplemented
```
makes it more readable

File: pilz_robot_programming/src/pilz_robot_programming/commands.py
Comment (ct2034) [inline pilz_robot_programming/src/pilz_robot_programming/commands.py]:
```suggestion
        return hash(tuple(map(lambda k: self.__dict__[k], sorted(self.__dict__.keys()))))
```
I did not test it, but this should make https://github.com/PilzDE/pilz_industrial_motion/pull/255/files#diff-288808388fd41c35fcc136672ed08a47R207-R213 not necessary

**Review by ct2034 — CHANGES_REQUESTED** [review]
These changes are super cool and definitely helpful. I have two suggestions in the code.
Generally, please also add unit tests, then we could see how my suggestions perform ;-)

**Comment by rfeistenauer** [comment]
Thank you for your reply. I waited with the tests until someone approved this change. Will provide them shortly in this case. Ill also look into your changes then.

File: pilz_robot_programming/src/pilz_robot_programming/commands.py
Comment (rfeistenauer) [reply pilz_robot_programming/src/pilz_robot_programming/commands.py]: I do not really agree, that adding extra stuff to read makes it more readable. Id rather just delete the empty line.

File: pilz_robot_programming/src/pilz_robot_programming/commands.py
Comment (rfeistenauer) [reply pilz_robot_programming/src/pilz_robot_programming/commands.py]: I had to change the method. Did try to change and remove/replace code like you suggested anyway, but I didn't find a working solution. Either provide more precise input, of what to delete or replace, or test it yourself with the provided tests please.

File: pilz_robot_programming/src/pilz_robot_programming/commands.py
Comment (rfeistenauer) [reply pilz_robot_programming/src/pilz_robot_programming/commands.py]: Is this with the new code still an issue?

File: pilz_robot_programming/src/pilz_robot_programming/commands.py
Comment (rfeistenauer) [inline pilz_robot_programming/src/pilz_robot_programming/commands.py:190]: Should probably throw an `NotImplementedError` in the furture, but didn't want to introduce an api break

File: pilz_robot_programming/src/pilz_robot_programming/commands.py
Comment (rfeistenauer) [inline pilz_robot_programming/src/pilz_robot_programming/commands.py:442]: This is already not sweet, and will get quite messy, with more types. We should try to find an alternative.

File: pilz_robot_programming/src/pilz_robot_programming/commands.py
Comment (rfeistenauer) [reply pilz_robot_programming/src/pilz_robot_programming/commands.py]: more or less implemented your change, you can check if you are OK with my implementation.

**Review by jschleicher — APPROVED** [review]
LGTM

**Comment by jschleicher** [comment]
@ct2034 Please Re-review or dismiss your old review.

**Review by ct2034 — APPROVED** [review]

**Pull request merged**
